cpp/cmake/slice2cpp.cmake

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) ZeroC, Inc.
+
+# Function to generate C++ source files from Slice (.ice) files for a target using slice2cpp
+# The target must have the Slice files in its sources
+# The generated files are added to the target sources
+# Usage:
+# add_executable(a_target source1.cpp source2.ice source3.ice)
+# slice2cpp_generate(a_target)
+function(slice2cpp_generate target)
```

Comment:
Did you think on how to handle Slice file dependencies to avoid rebuilds?

Reply:
No, I need to look into this.